src/common/sorted-table/index.js

```diff
@@ -560,8 +561,8 @@ export default class SortedTable extends Component {
     const { state } = this
     return handler(
       state.all
-        ? map(this._getItems(), 'id')
-        : state.selectedItemsIds.toArray()
+        ? this._getItems()
+        : filter(this._getItems(), item => state.selectedItemsIds.has(item.id))
```

Comment:
I'm not overly fond of this because the complexity is `O(number of items)` instead of `O(number of selected items)`, but I don't see a better approach…